community/kernel/src/main/java/org/neo4j/unsafe/batchinsert/BatchInserters.java

```diff
@@ -46,7 +47,7 @@
      * @param storeDir the store directory
      * @return a new {@link BatchInserter}
      */
-    public static BatchInserter inserter( String storeDir )
+    public static BatchInserter inserter( String storeDir ) throws IOException
```

Comment:
I have mixed feelings about this ... brakes compilation when upgrading 2.2 -> 2.3 due to this added checked exception. The IOException comes out of the constructor of _org.neo4j.kernel.impl.logging.StoreLogService#StoreLogService_ (well, from whatever he's doing). What meaningful thing is a client to do with an IOException when something around the logging fails other than crash? Isn't this specific exception more of a neo4j-internal detail, the logging service specifically?